phpBB/phpbb/mention/helper/mention_helper.php

```diff
@@ -0,0 +1,207 @@
+    */
+    public function get_mentioned_users($post_text)
+    {
+        $regular_expression_match = '#\[mention\](.*?)\[/mention\]#';
+        $matches = false;
+        $matches = $this->get_regex_match($regular_expression_match, $post_text);
+        if (count($matches[1]) > 0)
+        {
+            $start_tag_length = strlen('[mention]');
+            $end_tag_length = strlen('[\mention]');
+            $user_list = array();
+            $userid_list = array();
+            $user_list = $this->get_user_list($matches);
+            if (count($user_list) > 0)
+            {
+                $temp_notif_type_object = $this->notification->get_item_type_class('notification.type.mention');
+                $userid_list = $temp_notif_type_object->find_users_for_notification($this->db, $user_list);
+                if (count($userid_list) > 0)
+                {
+                    $new_post_data = $this->get_regex_substituted_text($matches, $post_text, $start_tag_length, $end_tag_length, $userid_list);
+                    return array('new_post_text' => $new_post_data['post_text'], 'users_mentioned' => $new_post_data['users_mapped'], 'notif_type_object' => $temp_notif_type_object);
+                }
+            }
+        }
+        return $post_text;
+    }
+
+    /**
+    * Function to generate Notifications.
+    *
+    * @param $user_list    array           Array containing userids to send
+    *                                       notifications to.
+    * @param $notif_manager_obj  \phpbb\notification\manager  Notification Manager object
+    * @param $temp_notif_type_object      object    Temporary notiifcation type object\
+    *                                                to call the method from
+    *                                                new notiifcation type - mention.
+    *
+    */
+    public function send_notifications($user_list, $temp_notif_type_object, $data)
+    {
+        $this->data = $data;
+        $notification_method_array = array();
+        $notification_details_list = $temp_notif_type_object->get_notification_type_and_method($this->db, $user_list, $notif_manager_obj);
+        $this->notification->add_notifications_for_users('notification.type.mention', $this->data, $notification_details_list);
+    }
+
+    /**
+    * Function to generate list of all users eligible to be tagged in posts.
+    *
+    * @param $keyword    string     Extract only users whose username matches keyword.
+    *
+    * @return $return_usernames_userid   array   Array containing user details matching
+    *                                            the keyword.
+    *
+    */
+    public function get_allusers($keyword)
+    {
+        $sql_query = 'SELECT user_id, username FROM ' . USERS_TABLE . ' WHERE user_id <> ' . ANONYMOUS . ' AND ' . $this->db->sql_in_set('user_type', [USER_NORMAL, USER_FOUNDER]) .  ' AND username_clean ' . $this->db->sql_like_expression($keyword . $this->db->get_any_char());
```

Comment:
Ok think about it, the best would be anyway a config setting to change the order and character starting (current >2). 
I think a query limit is important for huge forums and the order A-Z is there not working (only with much frustration).